bin/picom-trans

```diff
@@ -46,194 +54,324 @@ print_usage() {
-        ;;
-        name | window | opacity)
-          eval v=\$$OPTIND
-          OPTIND=$((OPTIND + 1))
-        ;;
-        name=* | window=* | opacity=*)
-          v=$(echo "$OPTARG" | sed -E 's/^[^=]+=//')
-        ;;
-        *)
-          echo "$0: illegal option $OPTARG" >& 2
-          exit 1
-        ;;
-      esac
-      option=$(echo "$OPTARG" | cut -c 1)
-      OPTARG=$v
-    fi
-    case "$option" in
-      h) print_usage; exit 0 ;;
-      s) wprefix=''; window='' ;;
-      c)
-        active=$(xprop -root -notype _NET_ACTIVE_WINDOW \
-          | grep -Eo '0x[[:xdigit:]]+' | head -n 1)
-        wprefix='-id'; window=$active
-      ;;
-      r) action='reset' ;;
-      t) action='toggle' ;;
-      d) action='delete' ;;
-      g) action='get' ;;
-      n) wprefix='-name'; window=$OPTARG ;;
-      w) wprefix='-id'; window=$OPTARG ;;
-      o) opacity=$OPTARG ;;
-      \?) break ;;
+    # Separate  "--ARG=VAL"  into  "--ARG" "VAL"
+    case "$1" in
+      (--opacity=*|--name=*|--window=*)
+        ARG="$(echo "$1" | sed -E 's/(--[^=]+)=.*$/\1/')"
+        VAL="${1##${ARG}=}"
+        shift && set -- "$ARG" "$VAL" "$@"
+    esac
+
+    # Turn into short form
+    case "$1" in
+      (--help|--opacity|--get|--delete|--toggle|--reset|--select|--current|--name|--window)
+        ARG=${1#-}                      # remove one '-' from prefix
+        ARG="$(echo $ARG | cut -c -2)"  # get first two characters
+        shift && set -- "$ARG" "$@"
     esac
+
+    # If the argument still starts with --, it is an invalid argument
+    case "$1" in
+      (--*)
+        echo "$EXE_NAME: illegal option $1" >&2
+        exit 1
+    esac
+
+    # Prepare for next iteration
+    ARG="$1"
+    shift && set -- "$@" "$ARG"
+    i=$(( i + 1 ))
   done
 
-  # Read positional arguments
-  shift $((OPTIND - 1))
-  test -n "$1" && opacity=$1 && shift
-done
 
-# clean up opacity. xargs == a poor man's trim.
-opacity=$(echo "$opacity" | xargs | sed 's/%//g')
+  while test $# -gt 0; do
+    # 15 +15 -15 are opacity values
+    # filter them out here, otherwise getopts will complain
+    if echo "$1" | grep -qE '^[+-]?[[:digit:]]+%?$'; then
+      target_opacity=$1
+      shift
+      continue
+    fi
 
-# Validate opacity value
-if test -z "$action" && ! echo "$opacity" | grep -qE '^[+-]?[0-9]+$'; then
-  echo "Invalid opacity specified: $opacity."
-  exit 1
-fi
+    OPTIND=1
+    while getopts 'ho:gdtrscn:w:00:11:22:33:44:55:66:77:88:99:' OPTION "$@"
+    # Originally, the above command was -
+    #   while getopts 'ho:gdtrscn:w:' OPTION "$@"
+    # But the numbers were added to account for opacity values (see below)
+    #
+    # Why is each number repeated twice?
+    # Because once it is specified with a ':' suffix, and once without.
+    # For e.g. the string '11:' accounts for two cases -
+    # 1) The '1:' accounts for -
+    #       picom-trans -c -10
+    #       picom-trans -c -15
+    # 2) The '1' accounts for -
+    #       picom-trans -c -1
```

Comment:
Oh my, I am not sure I like how hacky this is.

Reply:
Does 7ca6c54 feel less hackier?

:D

Reply:
Much better :+1: